File: src/main/java/no/nav/syfo/config/SykmeldingerConfig.java
package no.nav.syfo.config;

import no.nav.syfo.services.ws.*;
import no.nav.tjeneste.virksomhet.sykmelding.v1.*;
import no.nav.tjeneste.virksomhet.sykmelding.v1.meldinger.*;
import org.apache.cxf.frontend.ClientProxy;
import org.springframework.beans.factory.annotation.Value;
import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
import org.springframework.context.annotation.*;

import static java.util.Collections.singletonList;
import static no.nav.syfo.utils.OIDCUtil.leggTilOnBehalfOfOutInterceptorForOIDC;

@Configuration
public class SykmeldingerConfig {

    public static final String MOCK_KEY = "sykmelding.syfoservice.withmock";

    private SykmeldingV1 port;

    @SuppressWarnings("unchecked")
    @Bean
    @Primary
    @ConditionalOnProperty(value = MOCK_KEY, havingValue = "false", matchIfMissing = true)
    public SykmeldingV1 SykmeldingV1(@Value("${sykmelding.v1.url}") String serviceUrl) {
        SykmeldingV1 port = new WsClient<SykmeldingV1>().createPort(serviceUrl, SykmeldingV1.class, singletonList(new LogErrorHandler()));
        STSClientConfig.configureRequestSamlTokenOnBehalfOfOidc(port);
        this.port = port;
        return port;
    }

    public WSHentSykmeldingListeResponse hentSykmeldingListe(WSHentSykmeldingListeRequest request, String OIDCToken) throws HentSykmeldingListeSikkerhetsbegrensning {
        leggTilOnBehalfOfOutInterceptorForOIDC(ClientProxy.getClient(port), OIDCToken);
        return port.hentSykmeldingListe(request);
    }

    public WSHentOppfoelgingstilfelleListeResponse hentOppfoelgingstilfelleListe(WSHentOppfoelgingstilfelleListeRequest request, String OIDCToken) throws HentOppfoelgingstilfelleListeSikkerhetsbegrensning {
        leggTilOnBehalfOfOutInterceptorForOIDC(ClientProxy.getClient(port), OIDCToken);
        return port.hentOppfoelgingstilfelleListe(request);
    }
}

File: src/main/kotlin/no/nav/syfo/config/consumer/SykmeldingerConfig.kt
package no.nav.syfo.config.consumer

import no.nav.syfo.consumer.util.ws.*
import no.nav.syfo.oidc.OIDCUtil.leggTilOnBehalfOfOutInterceptorForOIDC
import no.nav.tjeneste.virksomhet.sykmelding.v1.HentOppfoelgingstilfelleListeSikkerhetsbegrensning
import no.nav.tjeneste.virksomhet.sykmelding.v1.SykmeldingV1
import no.nav.tjeneste.virksomhet.sykmelding.v1.meldinger.WSHentOppfoelgingstilfelleListeRequest
import no.nav.tjeneste.virksomhet.sykmelding.v1.meldinger.WSHentOppfoelgingstilfelleListeResponse
import org.apache.cxf.frontend.ClientProxy
import org.springframework.beans.factory.annotation.Value
import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty
import org.springframework.context.annotation.*
import javax.xml.ws.handler.Handler

@Configuration
class SykmeldingerConfig {
    private lateinit var port: SykmeldingV1

    @Bean
    @Primary
    @ConditionalOnProperty(value = [MOCK_KEY], havingValue = "false", matchIfMissing = true)
    fun SykmeldingV1(@Value("\${sykmelding.v1.url}") serviceUrl: String): SykmeldingV1 {
        val port = WsClient<SykmeldingV1>().createPort(serviceUrl, SykmeldingV1::class.java, listOf<Handler<*>>(LogErrorHandler()))
        STSClientConfig.configureRequestSamlTokenOnBehalfOfOidc(port)
        this.port = port
        return port
    }

    @Throws(HentOppfoelgingstilfelleListeSikkerhetsbegrensning::class)
    fun hentOppfoelgingstilfelleListe(request: WSHentOppfoelgingstilfelleListeRequest, OIDCToken: String): WSHentOppfoelgingstilfelleListeResponse {
        leggTilOnBehalfOfOutInterceptorForOIDC(ClientProxy.getClient(port), OIDCToken)
        return port.hentOppfoelgingstilfelleListe(request)
    }

    companion object {
        const val MOCK_KEY = "sykmelding.syfoservice.withmock"
    }
}
